sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UnnecessaryUsings.cs

```diff
@@ -65,32 +63,26 @@ protected override void Initialize(SonarAnalysisContext context)
                 },
                 SyntaxKind.CompilationUnit);
 
-        }
-
-        private static void VisitContent(CSharpRemovableUsingWalker visitor, SyntaxList<MemberDeclarationSyntax> members, IEnumerable<SyntaxTrivia> trivias)
+        private static void VisitContent(CSharpSyntaxWalker visitor, SyntaxList<MemberDeclarationSyntax> members, IEnumerable<SyntaxTrivia> trivias)
         {
-            var comments = trivias.Where(trivia => trivia.Kind() == SyntaxKind.SingleLineDocumentationCommentTrivia ||
-                trivia.Kind() == SyntaxKind.MultiLineDocumentationCommentTrivia);
+            var comments = trivias.Where(trivia => trivia.Kind() == SyntaxKind.SingleLineDocumentationCommentTrivia || trivia.Kind() == SyntaxKind.MultiLineDocumentationCommentTrivia);
 
             foreach (var member in members)
             {
                 visitor.SafeVisit(member);
             }
-            foreach (var comment in comments)
+            foreach (var comment in comments.Where(x => x.HasStructure))
             {
-                if (comment.HasStructure)
-                {
-                    visitor.SafeVisit(comment.GetStructure());
-                }
+                visitor.SafeVisit(comment.GetStructure());
             }
         }
 
-        private static void CheckUnnecessaryUsings(SyntaxNodeAnalysisContext context, IEnumerable<UsingDirectiveSyntax> usingDirectives, HashSet<INamespaceSymbol> necessaryNamespaces)
+        private static void CheckUnnecessaryUsings(SyntaxNodeAnalysisContext context, IEnumerable<UsingDirectiveSyntax> usingDirectives, IEnumerable<INamespaceSymbol> necessaryNamespaces)
```

Comment:
`necessaryNamespaces` can be iterated (and evaluated) multiple times since it is inside a foreach loop. We should let this be a HashSet

Reply:
`necessaryNamespaces` are a Symbols while `usedNamespace` are syntaxes. The `Any` is needed in this case.

We can improve it with some larger refactoring, but there's usually not more than 20 imports in a single class. I don't think the performance gain is worth it.

Reply:
I did not asked to remove the `Any` call 😄